src/master.py

```diff
@@ -141,12 +103,109 @@ def work():
     bootstrap.update_ec2_stack(stackname, context, concurrency='serial')
     return True
 
-# TODO: extract just the salt update part from `remaster`
 @debugtask
 @requires_aws_stack
 def update_salt(stackname):
-    current_master_stack = core.find_master_for_stack(stackname)
-    return remaster(stackname, current_master_stack)
+    "updates the installed version of salt to the version specified in the project file. see `docs/salt.md`"
+    # TODO: turn this into a decorator
+    import cfn
+    # start the machine if it's stopped
+    # you might also want to acquire a lock so alfred doesn't stop things or update things
+    cfn._check_want_to_be_running(stackname, 1)
+
+    # complete knowledge of a stack's particulars
+    context = context_handler.load_context(stackname)
+    if not context.get('ec2'):
+        LOG.info("no ec2 context: %s", stackname)
+        return
+
+    # 1. get the value of the desired version of salt from the project config file (pdata)
+    # 2. update the stack's current context and write that back to s3
+    # 3. update the subset of the context on the ec2 instance itself (buildvars)
+    # 4. run highstate
+    LOG.info("updating salt version in context file")
+    pdata = core.project_data_for_stackname(stackname)
+    context['project']['salt'] = pdata['salt']
+    context_handler.write_context(stackname, context)
+
+    LOG.info("updating buildvars on ec2 instance")
+    buildvars.refresh(stackname, context)
+
+    LOG.info("running highstate")
+    bootstrap.update_ec2_stack(stackname, context, concurrency='serial')
+
+    return True
+
+# this is untested and probably unsafe/buggy
+#@debugtask
+def update_salt_everywhere():
+
+    # lsh, 2019-03-22: below code is copied from remaster_all, which *has* been used before
+
+    assert False, "this is meant for our future selves as an example in bulk updating the salt version on all minions"
```

Comment:
I'll be ignoring anything after this line due to this comment, perhaps comment out if not used at the moment?